packages/mds-stream/nats/agency-stream-nats.ts

```diff
@@ -0,0 +1,23 @@
+import { VehicleEvent, Telemetry, Device } from '@mds-core/mds-types'
+import { getEnvVar } from '@mds-core/mds-utils'
+import { AgencyStreamInterface } from '../agency-stream-interface'
+import { NatsStreamProducer } from './stream-producer'
+
+const { TENANT_ID } = getEnvVar({
+  TENANT_ID: 'mds'
+})
+const deviceProducer = NatsStreamProducer<Device>(`${TENANT_ID}.device`)
+const eventProducer = NatsStreamProducer<VehicleEvent>(`${TENANT_ID}.event`)
+const telemetryProducer = NatsStreamProducer<Telemetry>(`${TENANT_ID}.telemetry`)
+
+export const AgencyStreamNats: AgencyStreamInterface = {
+  initialize: async () => {
+    await Promise.all([deviceProducer.initialize(), eventProducer.initialize(), telemetryProducer.initialize()])
+  },
+  writeEvent: eventProducer.write,
+  writeTelemetry: telemetryProducer.write,
+  writeDevice: deviceProducer.write,
+  shutdown: async () => {
+    await Promise.all([deviceProducer.shutdown(), eventProducer.shutdown(), telemetryProducer.shutdown()])
```

Comment:
This looks pretty clean, I do wonder what happens if one throws an exception before the others resolve though, will the other services still get shut down?

Reply:
`Promise.allSettled` is supported in Node 12 and TS with the ES2020 target.

Reply:
Indeed, we probably ought to be using a `Promise.allSettled()`

Reply:
No can haz `Promise.allSettled` with Node 12.14.1 :'( https://kangax.github.io/compat-table/es2016plus/